packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import uuid
+
+import requests
+
+from test_util.marathon import get_test_app
+from test_util.recordio import Decoder, Encoder
+
+
+# Wrapper to post, log, and validate return status
+def post(url, headers, json=None, data=None, stream=False):
+    r = requests.post(url, headers=headers, json=json, data=data, stream=stream)
+    logging.info(
+        'Got %s with POST request to %s with headers %s and json data %s.',
+        r.status_code,
+        url,
+        headers,
+        json
+    )
+    assert r.status_code == 200
+    return r
+
+
+# Creates and yields the initial ATTACH_CONTAINER_INPUT message, then a data message,
+# then an empty data chunk to indicate end-of-stream.
+def input_streamer(nested_container_id):
+    encoder = Encoder(lambda s: bytes(json.dumps(s, ensure_ascii=False), "UTF-8"))
+    message = {
+        'type': 'ATTACH_CONTAINER_INPUT',
+        'attach_container_input': {
+            'type': 'CONTAINER_ID',
+            'container_id': nested_container_id}}
+    yield encoder.encode(message)
+
+    message['attach_container_input'] = {
+        'type': 'PROCESS_IO',
+        'process_io': {
+            'type': 'DATA',
+            'data': {'type': 'STDIN', 'data': 'meow'}}}
+    yield encoder.encode(message)
+
+    # Place an empty string to indicate EOF to the server and push
+    # 'None' to our queue to indicate that we are done processing input.
+    message['attach_container_input']['process_io']['data']['data'] = ''
+    yield encoder.encode(message)
+
+
+def test_if_marathon_app_can_be_debugged(cluster):
+    # Launch a basic marathon app (no image), so we can debug into it!
+    # Cannot use deploy_and_cleanup because we must attach to a running app/task/container.
+    app, test_uuid = get_test_app()
+    app_id = 'integration-test-{}'.format(test_uuid)
+    cluster.marathon.deploy_app(app)
+
+    # Fetch the mesos master state once the task is running
+    master_state_url = 'http://{}:{}/state'.format(cluster.masters[0], 5050)
+    r = requests.get(master_state_url)
+    logging.debug('Got %s with request for %s. Response: \n%s', r.status_code, master_state_url, r.text)
+    assert r.status_code == 200
+    state = r.json()
+
+    # Find the agent_id and container_id from master state
+    container_id = None
+    agent_id = None
+    for framework in state['frameworks']:
+        for task in framework['tasks']:
+            if app_id in task['id']:
+                container_id = task['statuses'][0]['container_status']['container_id']['value']
+                agent_id = task['slave_id']
+    if container_id is None:
```

Comment:
All of these
```
if foo is None:
    raise Exception(my_message)
```
are suboptimal. When using pytest, always try to use plain asserts. The following is identical:
```
assert foo is None, my_message
```